model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro_program.py

```diff
@@ -297,13 +254,12 @@ def compute_perturbed_rho_and_potential_temperatures_at_half_and_full_levels(
     theta_v_ic: fa.CellKField[float],
     z_th_ddz_exner_c: fa.CellKField[float],
     k_field: fa.KField[gtx.int32],
-    nlev: gtx.int32,
     horizontal_start: gtx.int32,
     horizontal_end: gtx.int32,
     vertical_start: gtx.int32,
     vertical_end: gtx.int32,
 ):
-    _compute_perturbed_rho_and_potential_temperatures_at_half_and_full_levels(
+    _compute_pressure_gradient_and_perturbed_rho_and_potential_temperatures(
```

Comment:
Is renaming and removing `nlev` enough here? Do you not need to now call two different field operators with different domains>

Reply:
I actually do not know if I still need to split this stencil. I am now testing gpu run. I will respond to you again.

Reply:
So, in the end, it is not necessary to split this stencil. I renamed it because the new name reflects correctly the computation done in this combined stencil. As suggested by Christoph, we need to refactor the dycore for better readability including renaming and correctness in maybe next cycle.

Reply:
Ok sounds good!